eng/pipelines/templates/jobs/archetype-sdk-client.yml

```
@@ -6,249 +37,195 @@ jobs:
 
-      # Build with no dependencies at all (No samples) - PRECONDITIONS OFF
-      # WARNINGS_AS_ERRORS          ON
-      # TRANSPORT_CURL              OFF
-      # UNIT_TESTING                OFF
-      # UNIT_TESTING_MOCKS          OFF
-      # TRANSPORT_PAHO              OFF
-      # PRECONDITIONS               OFF
-      # LOGGING                     OFF
-      # AZ_PLATFORM_IMPL            AZ_PLATFORM_IMPL_NONE
-      Linux_ARM_no_preconditions:
+      LinuxARM:
         OSVmImage: 'ubuntu-18.04'
         vcpkg.deps: ''
+        VCPKG_DEFAULT_TRIPLET: 'x64-linux'
         AptDependencies: 'gcc-arm-none-eabi'
-        build.args: '-DCMAKE_TOOLCHAIN_FILE=cmake-modules/gcc-arm-toolchain.cmake -DPRECONDITIONS=OFF'
+        build.args: '-DCMAKE_TOOLCHAIN_FILE=cmake-modules/gcc-arm-toolchain.cmake -DPRECONDITIONS=OFF -DLOGGING=OFF'
         AZ_SDK_C_NO_SAMPLES: 'true'
+        PublishMapFiles: 'false'
+        BuildType: Debug
 
-      # Build with no dependencies at all (No samples) - PRECONDITIONS ON
-      # WARNINGS_AS_ERRORS          ON
-      # TRANSPORT_CURL              OFF
-      # UNIT_TESTING                OFF
-      # UNIT_TESTING_MOCKS          OFF
-      # TRANSPORT_PAHO              OFF
-      # PRECONDITIONS               ON
-      # LOGGING                     ON
-      # AZ_PLATFORM_IMPL            AZ_PLATFORM_IMPL_NONE
-      Linux_x64:
+      Linux_Release_MapFiles_Logging_UnitTests_Mocks:
         OSVmImage: 'ubuntu-18.04'
-        vcpkg.deps: ''
-        AZ_SDK_C_NO_SAMPLES: 'true'
-      Linux_x64_gcc5:
-        OSVmImage: 'ubuntu-16.04'
-        vcpkg.deps: ''
-        AZ_SDK_C_NO_SAMPLES: 'true'
-        CC: '/usr/bin/gcc-5'
-      Win_x86:
-        OSVmImage: 'windows-2019'
-        vcpkg.deps: ''
-        CMAKE_GENERATOR: 'Visual Studio 16 2019'
-        CMAKE_GENERATOR_PLATFORM: Win32
-        AZ_SDK_C_NO_SAMPLES: 'true'
-      Win_x64:
-        OSVmImage: 'windows-2019'
-        vcpkg.deps: ''
-        CMAKE_GENERATOR: 'Visual Studio 16 2019'
-        CMAKE_GENERATOR_PLATFORM: x64
+        vcpkg.deps: 'cmocka'
+        VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+        build.args: '-DPRECONDITIONS=OFF -DUNIT_TESTING=ON -DUNIT_TESTING_MOCKS=ON'
         AZ_SDK_C_NO_SAMPLES: 'true'
-      MacOS_x64:
-       OSVmImage: 'macOS-10.15'
-       vcpkg.deps: ''
-       AZ_SDK_C_NO_SAMPLES: 'true'
-
-      # Build with sample dependencies [curl for transport] - PRECONDITIONS ON
-      # WARNINGS_AS_ERRORS          ON
-      # TRANSPORT_CURL              ON
-      # UNIT_TESTING                OFF
-      # UNIT_TESTING_MOCKS          OFF
-      # TRANSPORT_PAHO              ON
-      # PRECONDITIONS               ON
-      # LOGGING                     ON
-      # AZ_PLATFORM_IMPL            AZ_PLATFORM_IMPL_POSIX / AZ_PLATFORM_IMPL_WIN32
-      Linux_x64_with_samples:
+        PublishMapFiles: 'true'
+        MapFileArtifactSuffix: 'lnx-x64-rel-noprc-log'
+        BuildType: Release
+
+      Linux_Logging_UnitTests_Mocks_Samples:
```

Same nit here and elsewhere (will stop mentioning):
```suggestion
      Linux_Debug_Logging_UnitTests_Mocks_Samples:
```